internal/commands/push/command_test.go

```diff
@@ -1429,6 +1431,89 @@ func TestPushCommandDeployDraftAndWait(t *testing.T) {
+	testError := errors.New("something bad happened")
+
+	t.Run("should return an error with a client that fails to import dependencies", func(t *testing.T) {
+		realmClient := mock.RealmClient{}
+
+		var capturedGroupID, capturedAppID, capturedUploadPath string
+		realmClient.ImportDependenciesFn = func(groupID, appID, uploadPath string) error {
+			capturedGroupID = groupID
+			capturedAppID = appID
+			capturedUploadPath = uploadPath
+			return testError
+		}
+		err := importDependenciesAndWait(nil, realmClient, appRemote{groupID, appID}, uploadPath)
+		assert.Equal(t, testError, err)
+
+		t.Log("and should properly pass through the expected inputs")
+		assert.Equal(t, groupID, capturedGroupID)
+		assert.Equal(t, appID, capturedAppID)
+		assert.Equal(t, uploadPath, capturedUploadPath)
+	})
+
+	t.Run("with a client that successfully imports dependencies", func(t *testing.T) {
+		realmClient := mock.RealmClient{}
+		realmClient.ImportDependenciesFn = func(groupID, appID, uploadPath string) error {
+			return nil
+		}
+
+		t.Run("but fails to finish the installation", func(t *testing.T) {
+			realmClient.DependenciesInstallationFn = func(groupID, appID string) (realm.DependenciesInstallation, error) {
+				return realm.DependenciesInstallation{}, testError
+			}
+
+			out, ui := mock.NewUI()
+			err := importDependenciesAndWait(ui, realmClient, appRemote{groupID, appID}, uploadPath)
+			t.Log("and should fail with error")
+			assert.Equal(t, err, testError)
+			assert.Equal(t, out.String(), "")
+		})
+
+		t.Run("and successfully finishes the installation", func(t *testing.T) {
+			var polls int
+
+			realmClient.DependenciesInstallationFn = func(groupID, appID string) (realm.DependenciesInstallation, error) {
+				status := realm.DependenciesStatusCreated
+				if polls > 1 {
+					status = realm.DependenciesStatusSuccessful
+				}
+				polls++
+				return realm.DependenciesInstallation{Status: status, StatusMessage: "processing"}, nil
+			}
+
+			out, ui := mock.NewUI()
+
+			t.Log("and should succeed with right output")
+			err := importDependenciesAndWait(ui, realmClient, appRemote{groupID, appID}, uploadPath)
+			assert.Nil(t, err)
+
+			assert.Equal(t, out.String(), "Dependencies installation complete\n")
+		})
+		t.Run("and eventually reaches unsuccessful installation", func(t *testing.T) {
+			var polls int
+
+			realmClient.DependenciesInstallationFn = func(groupID, appID string) (realm.DependenciesInstallation, error) {
+				status := realm.DependenciesStatusCreated
+				if polls > 1 {
+					status = realm.DependenciesStatusFailed
+				}
+				polls++
+				return realm.DependenciesInstallation{Status: status, StatusMessage: "processing"}, nil
+			}
+
+			out, ui := mock.NewUI()
+
+			t.Log("and should fail with installation error")
+			err := importDependenciesAndWait(ui, realmClient, appRemote{groupID, appID}, uploadPath)
+			assert.Equal(t, err, errors.New("failed to install dependencies"))
```

Comment:
also I'm just noticing this now, but with the test code here we actually want `assert.Equal(t, expected, actual)`

do you mind going through all your tests and here ensuring we use that order?  it just aligns better with some of the auto-generated test failure messages this way